addons/stock_account/models/product.py

```diff
@@ -370,25 +370,33 @@ def _run_fifo_vacuum(self, company=None):
         self.ensure_one()
         if company is None:
             company = self.env.company
+        product = self.with_company(company.id)
+        quantity_svl = product.quantity_svl
+        value_svl = product.value_svl
         svls_to_vacuum = self.env['stock.valuation.layer'].sudo().search([
             ('product_id', '=', self.id),
             ('remaining_qty', '<', 0),
             ('stock_move_id', '!=', False),
             ('company_id', '=', company.id),
         ], order='create_date, id')
         as_svls = []
+
+        domain = [
+            ('company_id', '=', company.id),
+            ('product_id', '=', self.id),
+            ('remaining_qty', '>', 0),
```

Comment:
You could add a create_date >= svls_to_vacuum[0].create_date. It should never happen but we never know.